parser/src/main/java/com/facebook/coresql/lint/LintingVisitor.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.lint;
+
+import com.facebook.coresql.parser.AstNode;
+import com.facebook.coresql.parser.SqlParserDefaultVisitor;
+import com.facebook.coresql.warning.CoreSqlWarning;
+import com.facebook.coresql.warning.WarningCollector;
+
+import java.util.List;
+
+public abstract class LintingVisitor
+        extends SqlParserDefaultVisitor
+{
+    private WarningCollector warningCollector;
+
+    public LintingVisitor(WarningCollector collector)
+    {
+        this.warningCollector = collector;
+    }
+
+    public abstract List<CoreSqlWarning> lint(AstNode node);
+
+    public WarningCollector getWarningCollector()
+    {
+        return warningCollector;
+    }
+
+    public void setWarningCollector(WarningCollector collector)
```

Comment:
Is there any reason that collector is mutable?

Reply:
Probably not -- I'll remove this setter and make it final.